File: n-ary-tree-level-order-traversal/n-ary-tree-level-order-traversal.py
"""
# Definition for a Node.
class Node(object):
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""

class Solution(object):
    def levelOrder(self, root):
        """
        :type root: Node
        :rtype: List[List[int]]
        """
        result = list()
        
        def getNode(node, level=0):
            """
            :type node: Node
            :type level: int
            :rtype: none
            """
            if (node != None):
                if len(result) < level + 1:
                    result.append([])
                result[level].append(node.val)
                for child in node.children:
                    getNode(child, level + 1)
        
        getNode(root)
        return result

File: n-ary-tree-level-order-traversal/n-ary-tree-level-order-traversal.js
/**
 * // Definition for a Node.
 * function Node(val,children) {
 *    this.val = val;
 *    this.children = children;
 * };
 */

/**
 * @param {Node|null} root
 * @return {number[][]}
 */
var levelOrder = function(root) {
    let result = [];
    const getNodes = (node, level = 0) => {
        if (node !== null) {
            if (result[level] === undefined) {
                result[level] = [node.val];
            } else {
                result[level].push(node.val)
            }
            for (let child of node.children) {
                getNodes(child, level + 1);
            }   
        }
        return;
    }
    getNodes(root);
    return result;
};